testing/trino-testing/src/main/java/io/trino/testing/BaseTestParquetComplexTypePredicatePushDown.java

```diff
@@ -34,16 +34,70 @@ public void testParquetOnlyNullsRowGroupPruning()
         // Nested column `b` also has nulls count of 4096, but it contains non nulls as well
         assertUpdate("CREATE TABLE " + tableName + " (col ROW(a BIGINT, b ARRAY(DOUBLE))) WITH (format = 'PARQUET')");
         assertUpdate("INSERT INTO " + tableName + " SELECT * FROM unnest(transform(repeat(1, 4096), x -> ROW(ROW(NULL, ARRAY [NULL, rand()]))))", 4096);
-        // TODO replace with assertNoDataRead after nested column predicate pushdown
+
+        assertNoDataRead("SELECT * FROM " + tableName + " WHERE col.a IS NOT NULL");
+
+        assertQueryStats(
+                getSession(),
+                "SELECT * FROM " + tableName + " WHERE col.b IS NOT NULL",
+                queryStats -> assertThat(queryStats.getProcessedInputDataSize().toBytes()).isGreaterThan(0),
+                results -> assertThat(results.getRowCount()).isEqualTo(4096));
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testParquetNestedRowGroupPruning()
+    {
+        String tableName = "test_nested_column_pruning_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (col1Row ROW(a BIGINT, b BIGINT), col2Array ARRAY(BIGINT),  col3Map Map(VARCHAR, BIGINT)) WITH (format = 'PARQUET')");
+        assertUpdate("INSERT INTO " + tableName + " SELECT * FROM unnest(transform(SEQUENCE(1, 10000), x -> ROW(ROW(x*2, 100), ARRAY[100, 200], MAP(ARRAY['FOO', 'BAR'], ARRAY[100, 200]))))", 10000);
+
+        // no data read since the row dereference predicate is pushed down
+        assertNoDataRead("SELECT * FROM " + tableName + " WHERE col1Row.a = -1");
+
+        // read all since predicate case matches with the data
+        assertQueryStats(
+                getSession(),
+                "SELECT * FROM " + tableName + " WHERE col1Row.b = 100",
+                queryStats -> assertThat(queryStats.getProcessedInputDataSize().toBytes()).isGreaterThan(0),
+                results -> assertThat(results.getRowCount()).isEqualTo(10000));
+
+        // no predicate push down for array dereference
```

Comment:
Please move array tests to a different method and keep this method only for row type

Reply:
done.